File: 2021/10/08/main.go
package main

import "strings"

type Trie struct {
	words []string
}

func Constructor() Trie {
	return Trie{words: make([]string, 0)}
}

func (this *Trie) Insert(word string) {
	this.words = append(this.words, word)
}

func (this *Trie) Search(word string) bool {
	for _, w := range this.words {
		if w == word {
			return true
		}
	}
	return false
}

func (this *Trie) StartsWith(prefix string) bool {
	for _, w := range this.words {
		if strings.HasPrefix(w, prefix) {
			return true
		}
	}
	return false
}

/**
 * Your Trie object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Insert(word);
 * param_2 := obj.Search(word);
 * param_3 := obj.StartsWith(prefix);
 */

File: 2021/10/08/main.py
class Trie:

    def __init__(self):
        self.words = []

    def insert(self, word: str) -> None:
        self.words.append(word)
        

    def search(self, word: str) -> bool:
        for w in self.words:
            if w == word:
                return True
        return False

    def startsWith(self, prefix: str) -> bool:
        for w in self.words:
            if w.startswith(prefix):
                return True
        return False


# Your Trie object will be instantiated and called as such:
# obj = Trie()
# obj.insert(word)
# param_2 = obj.search(word)
# param_3 = obj.startsWith(prefix)